schema/object_test.go

```diff
@@ -426,6 +426,75 @@ func TestTypedObjectSchema_Any(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestDefaultsStructSerialization(t *testing.T) {
+	type TestData struct {
+		Foo *string `json:"foo"`
+	}
+	s := schema.NewTypedObject[TestData](
+		"TestData",
+		map[string]*schema.PropertySchema{
+			"foo": schema.NewPropertySchema(
+				schema.NewStringSchema(nil, nil, nil),
+				nil,
+				false,
+				nil,
+				nil,
+				nil,
+				schema.PointerTo(`"abc"`),
+				nil,
+			),
+		},
+	)
+	// First, unserialization
+	unserialized, err := s.Unserialize(map[string]any{})
+	assert.NoError(t, err)
+	assert.NotNil(t, unserialized)
+	assert.InstanceOf[TestData](t, unserialized)
+	assert.NotNil(t, unserialized.(TestData).Foo)
+	// Validate that default is included
+	assert.Equals(t, *unserialized.(TestData).Foo, "abc")
+
+	// Next, serialization.
+	serialized, err := s.Serialize(unserialized)
+	assert.NoError(t, err)
+	assert.NotNil(t, serialized)
+	assert.InstanceOf[map[string]any](t, serialized)
+	assert.Equals(t, assert.MapContainsKey[string](t, "foo", serialized.(map[string]any)), "abc")
+}
```

Comment:
Is line 462 coded as intended?  To my unpracticed eye, it looks like you're passing the result of one assertion as the argument to another, when it seems more likely that there should be two separate assertions, here (i.e., a check that the map contains the key and that the value of that key is `"abc"`).

Ditto line 495.

Reply:
The MapContainsKey assertion returns the value at the key. So it's two assertions in one line.
Does anyone think that it's worth changing this to two separate lines?

Reply:
Well, since the line is over 100 characters long, it probably should at least be _wrapped_, and if you're going to wrap it, you might as well break it into two separate assertions.  (You're just lucky that this is GoLang and not Python...otherwise, you'd have to take it up with Black!  😆) 

(Seriously, though, I don't know what it looks like when these assertions fail, but having two assertions in the same line means that the engineer looking at the failure has to separate them to understand what is going on, and that would be less of an issue if they were on separate lines....)

Reply:
Sure. I'll split it.
The messages are fairly clear about what the problems are. I don't think that part will be an issue.